### PR TITLE
Decode WebTransport streams' head

### PR DESCRIPTION
**Pull request by ddragana**

### CODE REVIEW

File: neqo-http3/src/connection.rs [hunk 1]
```diff
@@ -476,6 +471,53 @@ impl Http3Connection {
         self.recv_streams.clear();
     }
 
+    fn handle_stream_manipulation_output<U>(
```
Comment (KershawChang) [inline neqo-http3/src/connection.rs:474]: Sorry, I found this function is not easy to understand for me. Could you add some comments?

File: neqo-http3/src/client_events.rs [hunk 2]
```diff
@@ -83,24 +81,19 @@ pub struct Http3ClientEvents {
...
-    fn header_ready(&self, stream_id: u64, headers: Vec<Header>, interim: bool, fin: bool) {
-        self.insert(Http3ClientEvent::HeaderReady {
-            stream_id,
-            headers,
-            interim,
-            fin,
-        });
-    }
-
+impl RecvStreamEvents for Http3ClientEvents {
```
Comment (KershawChang) [inline neqo-http3/src/client_events.rs:84]: Sorry for a stupid question: is `RecvStreamEvents` only used by webtransport? what's the difference with `HttpRecvStreamEvents`?

Comment (ddragana) [reply neqo-http3/src/client_events.rs:84]: `RecvStreamEvents` contain functions that are more general and common for http streams and webtransport streams.
`HttpRecvStreamEvents` extends `RecvStreamEvents` and adds functions that are only specific to http streams

File: neqo-http3/src/connection.rs (hunk 1 above)
Comment (ddragana) [reply neqo-http3/src/connection.rs:489]: I have added some explanations.

**Review by KershawChang — APPROVED** [review]

**Pull request merged**